### ISSUE TITLE
Using only typedefs, not resolvers

### ISSUE THREAD

**Issue by BramKaashoek** [19majkel94/type-graphql#339]
I would like to use only the typeDef generating capabilities of this library without using the resolvers, since I prefer the apollo-style resolvers. https://github.com/19majkel94/type-graphql/issues/222 makes me think it should be possible, but I could not find the way to do it in the documentation. Could you point me to which function to use?

As far as I can see, the `buildTypeDefsAndResolvers` function as described in https://github.com/19majkel94/type-graphql/pull/233 still requires me to write (minimal) resolvers?

Is there a function such as  `buildTypeDefs` where I could pass in an array of my ObjectType/InputType classes to generate typeDefs?

**Comment by 19majkel94** [comment]
TypeGraphQL is all focused on resolvers and runtime improvements. Using it only for object and input typedefs is like driving your car without shifting the gears.

You can use simple code generator that will read your interfaces/classes and produce the typedefs SDL or maybe even try the reflection PoC:
https://github.com/19majkel94/typegraphql-reflection-poc

**Comment by BramKaashoek** [comment]
Thanks for the quick response. 

Is https://github.com/19majkel94/typegraphql-reflection-poc MIT licensed?

**Comment by BramKaashoek** [comment]
I wrote a small package to fill this need:
https://www.npmjs.com/package/typescript-typedefs

**Comment by 19majkel94** [comment]
I am reviving this feature as it might be handy to provide a step-by-step migration path and make use of the enhanced reflection plugin 💪

**Comment by syffs** [comment]
@BramKaashoek [this](https://www.npmjs.com/package/typescript-typedefs) seems nice, but it'd be great to handle `@InterfaceType()` to allow usage of typescript inheritence. What do you think ?

@19majkel94 this feature seems vital, there are plenty of scenarios which requires to work on typeDefs first without considering resolvers

**Comment by BramKaashoek** [comment]
> @BramKaashoek [this](https://www.npmjs.com/package/typescript-typedefs) seems nice, but it'd be great to handle `@InterfaceType()` to allow usage of typescript inheritence. What do you think ?

You're completely right, I'll add that.

**Comment by 19majkel94** [comment]
> Is this complex to implement ?

Yes, it's a requirement of `graphql-js` to have at least one query to build the schema.

I would have to rewrite the core to just do decorators/reflection -> metadata -> schema string magic - will do in some time in the future with a more modular codebase 😉

**Comment by syffs** [comment]
@BramKaashoek all things considered, this'd be perfect with circular references out of the box ! The [workaround](https://github.com/BramKaashoek/typescript-typedefs#why-wont-circular-references-work) is really not convenient...

Any idea on how to work around this ? I'm not aware of the root cause: would a PR in reflect-[metadata](https://github.com/rbuckton/reflect-metadata) help ?

**Comment by BramKaashoek** [comment]
@syffs After doing some more looking around I found 
[forward-ref](https://github.com/angular/angular/blob/master/packages/core/src/di/forward_ref.ts) as used by Angular, this might be a workaround. I'll look into it.

**Comment by syffs** [comment]
@BramKaashoek this might help, there's a suggestion on the [related issue on typescript repo](https://github.com/microsoft/TypeScript/issues/4521#issuecomment-384972777)

**Comment by syffs** [comment]
@BramKaashoek it's much more convenient with forwrad-ref !

> Edit: Added in version 0.1.9 https://www.npmjs.com/package/typescript-typedefs

A suggestion about Interface and inheritence, this might make sense to automatically cascade types from extended types / interfaces, or you end up with redundant `Field`, as below with `author`:
```graphql
@Interface()
class Book {
  @Field()
  author: string;
}

// option 1 for implementing
@Type()
class CourseBook extends Book {
  @Field()
  author: string;

  @Field()
  course: string;
}
```
what do you think ?

**Comment by BramKaashoek** [comment]
> A suggestion about Interface and inheritence, this might make sense to automatically cascade fields from extended types / interfaces, or you end up with redundant `Field`, as below with `author`:

This is what I was initially hoping to do, but sadly this is not possible due to how decorators work.

Let's take further discussion over to https://github.com/BramKaashoek/typescript-typedefs so that we don't bother Majkel as much :).
